oauth2_provider/oauth2_backends.py

```diff
@@ -180,7 +180,11 @@ def extract_body(self, request):
         :return: provided POST parameters "urlencodable"
         """
         try:
-            body = json.loads(request.body.decode('utf-8')).items()
+            data = json.loads(request.body.decode('utf-8'))
+            if isinstance(data, dict):
+                body = data.items()
+            else:
+                body = ""
```

Comment:
Why do you set an empty string if not getting a dictionary?

Reply:
@menecio `except AttributeError, ValueError:` is a syntax error.